packages/ast-spec/src/special/ImportSpecifier/spec.ts

```diff
@@ -2,9 +2,10 @@ import type { AST_NODE_TYPES } from '../../ast-node-types';
 import type { BaseNode } from '../../base/BaseNode';
 import type { ImportKind } from '../../declaration/ExportAndImportKind';
 import type { Identifier } from '../../expression/Identifier/spec';
+import type { StringLiteral } from '../../expression/literal/StringLiteral/spec';
 
 export interface ImportSpecifier extends BaseNode {
-  imported: Identifier;
+  imported: Identifier | StringLiteral;
```

Comment:
So there is one problem with this definition:
```ts
import { "😭" } from 'module';
```

This is parser valid TS code, but is invalid ESTree AST code.

So we need to ban this in the converter.

Reply:
Is it valid in the TS parser? I'm getting errors. https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgIkLwbhaPeXAvnAZlBCHAOQgQAmArgDYCmJA3EA

Reply:
You're getting a *_SEMANTIC_* error, not a *_SYNTACTIC_* error.
The code parses fine and then TS emits a semantic error.

So we need to check for this case and throw our own error.